src/main/java/com/exasol/adapter/metadata/converter/SchemaMetadataJsonConverter.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    /**
+     * Convert the given schema metadata to its JSON representation
+     *
+     * @param schemaMetadata schemaMetadata to be converted into JSON
+     * @return JSON representation
+     */
+    public JsonObject convert(final SchemaMetadata schemaMetadata) {
+        final JsonObjectBuilder root = this.factory.createObjectBuilder();
+        final JsonArrayBuilder tablesBuilder = this.factory.createArrayBuilder();
+        for (final TableMetadata table : schemaMetadata.getTables()) {
+            tablesBuilder.add(convertTableMetadata(table));
+        }
+        root.add(TABLES_KEY, tablesBuilder);
+        root.add(ADAPTER_NOTES_KEY, schemaMetadata.getAdapterNotes());
+        return root.build();
+    }
+
+    private JsonObjectBuilder convertTableMetadata(final TableMetadata table) {
+        final JsonObjectBuilder tableBuilder = this.factory.createObjectBuilder();
+        tableBuilder.add(TABLE_TYPE_KEY, "table");
+        tableBuilder.add(TABLE_NAME_KEY, table.getName());
+        final JsonArrayBuilder columnsBuilder = Json.createArrayBuilder();
+        for (final ColumnMetadata column : table.getColumns()) {
+            columnsBuilder.add(convertColumnMetadata(column));
+        }
+        if (table.hasAdapterNote()) {
+            tableBuilder.add(ADAPTER_NOTES_KEY, table.getAdapterNotes());
+        }
+        if (table.hasComment()) {
+            tableBuilder.add(COMMENT_KEY, table.getComment());
+        }
+        tableBuilder.add("columns", columnsBuilder);
+        return tableBuilder;
+    }
+
+    private JsonObjectBuilder convertColumnMetadata(final ColumnMetadata column) {
+        final JsonObjectBuilder columnBuilder = this.factory.createObjectBuilder();
+        columnBuilder.add(TABLE_NAME_KEY, column.getName());
+        columnBuilder.add(ADAPTER_NOTES_KEY, column.getAdapterNotes());
+        columnBuilder.add(DATA_TYPE_KEY, convertType(column.getType()));
+        if (!column.isNullable()) {
+            columnBuilder.add(NULLABLE_KEY, false);
+        }
+        if (column.isIdentity()) {
+            columnBuilder.add(IDENTITY_KEY, true);
+        }
+        if (column.hasDefault()) {
+            columnBuilder.add(DEFAULT_KEY, column.getDefaultValue());
+        }
+        if (column.hasComment()) {
+            columnBuilder.add(COMMENT_KEY, column.getComment());
+        }
+        return columnBuilder;
+    }
+
+    /**
+     * Convert a data type to its JSON representation
+     *
+     * @param dataType data type to be converted
+     * @return JSON representation
+     */
+    public JsonObject convertType(final DataType dataType) {
+        final JsonObjectBuilder typeAsJson = this.factory.createObjectBuilder();
+        typeAsJson.add(DATA_TYPEKEY, getExasolDataTypeName(dataType.getExaDataType()));
+        switch (dataType.getExaDataType()) {
+        case UNSUPPORTED:
+            throw new IllegalArgumentException("Unsupported Data Type, should never happen");
```

Comment:
I don't really like this error message...

Reply:
Improved all error messages in that class.